From 8ef3fe7c02df569084c54b37f273bb8651427410 Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.de>
Date: Thu, 7 Mar 2013 17:31:18 +0100
Subject: [PATCH 5/5] modify hub to detect unplugs in all states
References: bnc#807560
Patch-Mainline: Never 

This is necessary because we haven't taken the port handling patches

Signed-off-by: Oliver Neukum <oneukum@suse.de>
---
 drivers/usb/core/hub.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/core/hub.c b/drivers/usb/core/hub.c
index 08e3ce8..066cf57 100644
--- a/drivers/usb/core/hub.c
+++ b/drivers/usb/core/hub.c
@@ -3802,7 +3802,7 @@ static void hub_events(void)
 			if (hub_port_warm_reset_required(hub, portstatus)) {
 				int status;
 				struct usb_device *udev =
-					hub->ports[i - 1]->child;
+					hdev->children[i - 1];
 
 				dev_dbg(hub_dev, "warm reset port %d\n", i);
 				if (!udev) {
-- 
1.7.10.4

